hub/core/storage/provider.py

```diff
@@ -1,32 +1,42 @@
 from collections.abc import MutableMapping
 from typing import Optional
 from abc import ABC, abstractmethod
+from hub.util.assert_byte_indexes import assert_byte_indexes
+from hub.constants import BYTE_PADDING
 
 
-class Provider(ABC, MutableMapping):
+class StorageProvider(ABC, MutableMapping):
     """An abstract base class for implementing a provider.
 
     To add a new provider using Provider, create a subclass and implement all 5 abstract methods below.
-    Alternatively, you can inherit from MappedProvider and have a simpler implementation.
     """
 
     @abstractmethod
-    def __getitem__(
+    def __getitem__(self, path: str):
+        """Gets the object present at the path within the given byte range.
```

Comment:
byte range? there's no arg here

Reply:
it's shifted to get_bytes that calls getiem